### PR TITLE
Fix execution graph

### PR DESCRIPTION
**Pull request by blindpirate**
### Context

This fixes https://github.com/gradle/gradle-private/issues/2549 by refactoring javascript code. Previously the execution graph is rendered directly by `$.plot`, now it's using the `renderGraph` method so zoom-in/zoom-out are supported.

Demo: https://builds.gradle.org/repository/download/Gradle_Check_PerformanceTestCoordinator/25497748:id/report-performance-performance-tests.zip%21/report/tests/clean-assemble-on-largeMonolithicJavaProject-with-remote-https-cache.html?redirectSupported=false

### CODE REVIEW

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/Graph.java [hunk 1]
```diff
@@ -25,30 +25,33 @@
 import java.util.stream.IntStream;
 
 public class Graph {
```
Comment (wolfs) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/Graph.java]: Looks like `Graph` should be called `ExecutionGraph` or something similar.

File: subprojects/internal-performance-testing/src/main/resources/org/gradle/reporting/performanceGraph.js [hunk 2]
```diff
@@ -52,20 +62,11 @@
...
-                        }
-                    }
-            },
+            xaxis: xaxis,
             yaxis: { min: determineMinY(data, unit) }, selection: { mode: 'xy' } };
         var chart = $.plot('#' + chartId, data, options);
         plots[chartId] = chart;
-        var zoomFunction = function(plot, reset) {
+        function zoomFunction(plot, reset) {
             var reset = reset || false;
```
Comment (wolfs) [inline subprojects/internal-performance-testing/src/main/resources/org/gradle/reporting/performanceGraph.js]: `reset` is declared twice. Can you maybe change the parameter name?

File: subprojects/internal-performance-testing/src/main/resources/org/gradle/reporting/performanceGraph.js [hunk 3]
```diff
@@ -83,30 +84,45 @@
                 plot.clearSelection();
             };
         };
+
+        function hoverOnHistoryGraph(event, pos, item) {
+            var executionLabel = executionLabels[item.datapoint[0]];
+            var revLabel;
+            if (item.series.label == executionLabel.branch) {
```
Comment (wolfs) [inline subprojects/internal-performance-testing/src/main/resources/org/gradle/reporting/performanceGraph.js]: Use `===` for comparison.

File: subprojects/internal-performance-testing/src/main/resources/org/gradle/reporting/performanceGraph.js [hunk 4]
```diff
@@ -83,30 +84,45 @@
                 plot.clearSelection();
             };
         };
+
+        function hoverOnHistoryGraph(event, pos, item) {
+            var executionLabel = executionLabels[item.datapoint[0]];
```
Comment (wolfs) [inline subprojects/internal-performance-testing/src/main/resources/org/gradle/reporting/performanceGraph.js]: Use `const`.

File: subprojects/internal-performance-testing/src/main/resources/org/gradle/reporting/performanceGraph.js [hunk 5]
```diff
@@ -83,30 +84,45 @@
...
-                }
-                var text = revLabel + ', date: ' + executionLabel.date + ', '+ label + ': ' + item.datapoint[1] + unit;
-                $('#tooltip').html(text).css({top: item.pageY - 10, left: item.pageX + 10}).show();
+                hoverOnExecutionGraph(event, pos, item);
             }
         }).bind('plotselected', zoomFunction(chart)).bind('dblclick', zoomFunction(chart, true))
             .bind("plotclick",
                 function (event, pos, item) {
+                    if (!item) {
+                        // not select a plot
```
Comment (wolfs) [inline subprojects/internal-performance-testing/src/main/resources/org/gradle/reporting/performanceGraph.js]: Nit:
```suggestion
                        // no plot selected
```

**Comment by blindpirate** [comment]
Thanks @wolfs !

**Pull request merged**